AccessControlException creating EJB Timers

>------Start of DE processing------ = [4/3/19, 10:22:55:837 UTC]
>Exception = java.lang.reflect.InvocationTargetException
>Source = com.ibm.ws.serialization.DeserializationObjectInputStream
>probeid = 50
>Stack Dump = java.lang.reflect.InvocationTargetException
>	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
>	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
>	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
>	at com.ibm.ws.serialization.DeserializationObjectInputStream.<clinit>(DeserializationObjectInputStream.java:49)
>	at com.ibm.ws.serialization.internal.DeserializationContextImpl.createObjectInputStream(DeserializationContextImpl.java:41)
>	at com.ibm.ws.serialization.internal.SerializationServiceImpl.createObjectInputStream(SerializationServiceImpl.java:162)
>	at com.ibm.ws.ejbcontainer.osgi.internal.ObjectCopierImpl.copySerializable(ObjectCopierImpl.java:52)
>	at com.ibm.ws.ejbcontainer.util.ObjectCopier.copy(ObjectCopier.java:134)
>	at com.ibm.ws.ejbcontainer.runtime.AbstractEJBRuntime.createTimer(AbstractEJBRuntime.java:1809)
>	at com.ibm.ejs.container.BeanO.createCalendarTimer(BeanO.java:2086)
>	at suite.r80.base.timer.cal.ejb.NextTimeoutBean.createCalTimerWithSE(NextTimeoutBean.java:561)
>	at suite.r80.base.timer.cal.ejb.NextTimeoutBean.testHour(NextTimeoutBean.java:293)
>	at suite.r80.base.timer.cal.ejb.EJSLocal0SLNextTimeoutBean_305ff64f.testHour(EJSLocal0SLNextTimeoutBean_305ff64f.java)
>	at suite.r80.base.timer.cal.NextTimeoutServlet.testHour(NextTimeoutServlet.java:96)
>	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
>	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
>	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
>	at componenttest.app.FATServlet.doGet(FATServlet.java:67)
>	at javax.servlet.http.HttpServlet.service(HttpServlet.java:687)
>	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
>	at com.ibm.ws.webcontainer.servlet.ServletWrapper.service(ServletWrapper.java:1255)
>	at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:743)
>	at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:440)
>	at com.ibm.ws.webcontainer.filter.WebAppFilterManager.invokeFilters(WebAppFilterManager.java:1221)
>	at com.ibm.ws.webcontainer.filter.WebAppFilterManager.invokeFilters(WebAppFilterManager.java:1005)
>	at com.ibm.ws.webcontainer.servlet.CacheServletWrapper.handleRequest(CacheServletWrapper.java:75)
>	at com.ibm.ws.webcontainer.WebContainer.handleRequest(WebContainer.java:923)
>	at com.ibm.ws.webcontainer.osgi.DynamicVirtualHost$2.run(DynamicVirtualHost.java:279)
>	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink$TaskWrapper.run(HttpDispatcherLink.java:1047)
>	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.wrapHandlerAndExecute(HttpDispatcherLink.java:417)
>	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.ready(HttpDispatcherLink.java:376)
>	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleDiscrimination(HttpInboundLink.java:532)
>	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleNewRequest(HttpInboundLink.java:466)
>	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.processRequest(HttpInboundLink.java:331)
>	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.ready(HttpInboundLink.java:302)
>	at com.ibm.ws.tcpchannel.internal.NewConnectionInitialReadCallback.sendToDiscriminators(NewConnectionInitialReadCallback.java:165)
>	at com.ibm.ws.tcpchannel.internal.NewConnectionInitialReadCallback.complete(NewConnectionInitialReadCallback.java:74)
>	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.requestComplete(WorkQueueManager.java:503)
>	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.attemptIO(WorkQueueManager.java:573)
>	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.workerRun(WorkQueueManager.java:954)
>	at com.ibm.ws.tcpchannel.internal.WorkQueueManager$Worker.run(WorkQueueManager.java:1043)
>	at com.ibm.ws.threading.internal.ExecutorServiceImpl$RunnableWrapper.run(ExecutorServiceImpl.java:239)
>	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
>	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
>	at java.base/java.lang.Thread.run(Thread.java:825)
>Caused by: java.security.AccessControlException: Access denied ("java.lang.RuntimePermission" "getClassLoader")
>	at java.base/java.security.AccessController.throwACE(AccessController.java:176)
>	at java.base/java.security.AccessController.checkPermissionHelper(AccessController.java:237)
>	at java.base/java.security.AccessController.checkPermission(AccessController.java:373)
>	at java.base/java.lang.SecurityManager.checkPermission(SecurityManager.java:322)
>	at java.base/java.lang.ClassLoader.getPlatformClassLoader(ClassLoader.java:871)
]
	at componenttest.custom.junit.runner.FATRunner.blowup(FATRunner.java:294)
	at componenttest.custom.junit.runner.FATRunner.access$500(FATRunner.java:65)
	at componenttest.custom.junit.runner.FATRunner$1.evaluate(FATRunner.java:244)
	at componenttest.custom.junit.runner.OnlyRunInJava7Rule$1.evaluate(OnlyRunInJava7Rule.java:26)
	at componenttest.custom.junit.runner.FATRunner$2.evaluate(FATRunner.java:334)
	at componenttest.custom.junit.runner.FATRunner.run(FATRunner.java:168)

--- a/dev/com.ibm.ws.serialization/src/com/ibm/ws/serialization/DeserializationObjectInputStream.java
+++ b/dev/com.ibm.ws.serialization/src/com/ibm/ws/serialization/DeserializationObjectInputStream.java
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2012, 2017 IBM Corporation and others.
+ * Copyright (c) 2012, 2019 IBM Corporation and others.
  * All rights reserved. This program and the accompanying materials
  * are made available under the terms of the Eclipse Public License v1.0
  * which accompanies this distribution, and is available at
@@ -40,20 +40,7 @@ public class DeserializationObjectInputStream extends ObjectInputStream {
     private final ClassLoader classLoader;
 
     // The PlatformClassloader. It is set when running with java 9 and above.
-    private static final ClassLoader platformClassloader;
-    static {
-        ClassLoader pcl = null;
-        if (JavaInfo.majorVersion() >= 9) {
-            try {
-                Method getPlatformClassLoader = ClassLoader.class.getMethod("getPlatformClassLoader");
-                pcl = (ClassLoader) getPlatformClassLoader.invoke(null);
-            } catch (Throwable t) {
-                // Log an FFDC.
-            }
-        }
-
-        platformClassloader = pcl;
-    }
+    private static final ClassLoader platformClassloader = getPlatformClassLoader();
 
     public DeserializationObjectInputStream(InputStream in, ClassLoader classLoader) throws IOException {
         super(in);
@@ -260,5 +247,27 @@ public class DeserializationObjectInputStream extends ObjectInputStream {
                 return klass.getClassLoader();
             }
         });
+    }
+
+    /**
+     * Returns the PlatformClassloader when running with java 9 and above; otherwise returns null.
+     */
+    private static ClassLoader getPlatformClassLoader() {
+        if (JavaInfo.majorVersion() >= 9) {
+            return AccessController.doPrivileged(new PrivilegedAction<ClassLoader>() {
+                @Override
+                public ClassLoader run() {
+                    ClassLoader pcl = null;
+                    try {
+                        Method getPlatformClassLoader = ClassLoader.class.getMethod("getPlatformClassLoader");
+                        pcl = (ClassLoader) getPlatformClassLoader.invoke(null);
+                    } catch (Throwable t) {
+                        // Log an FFDC.
+                    }
+                    return pcl;
+                }
+            });
+        }
+        return null;
     }
 }

